src/AdvancedCircuitBreaker.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\CircuitBreaker;
+
+use PrestaShop\CircuitBreaker\Contracts\Client;
+use PrestaShop\CircuitBreaker\Contracts\ConfigurableCall;
+use PrestaShop\CircuitBreaker\Contracts\Storage;
+use PrestaShop\CircuitBreaker\Contracts\System;
+use PrestaShop\CircuitBreaker\Contracts\Transitioner;
+use PrestaShop\CircuitBreaker\Exceptions\UnavailableServiceException;
+
+/**
+ * This implementation of the CircuitBreaker is a bit more advanced than the SimpleCircuitBreaker,
```

Comment:
I'm wondering: if you're bumping a new major, how about adding parameters as part of the CircuitBreaker contract? => https://github.com/loveOSS/resiliency/blob/master/src/Contracts/CircuitBreaker.php#L26

Also, as this library will be integrated starting 1.7.7 don't you think you could also bump PHP requirements to PHP 7 and so on... use resiliency instead? :angel:

Reply:
I also wanted to change this.. this `callWithParameters` method bothers me. Ok I'll change it
As for the minimum version for PHP this will be used in modules for 176 so we can't change it for now

Reply:
done

Reply:
> don't you think you could also bump PHP requirements to PHP 7 and so on...

Don't! This must be usable by module developers who need to go all the way to 1.6